src/main/java/org/folio/service/request/RequestBatchResourceService.java

```diff
@@ -85,24 +85,29 @@ public void executeRequestBatchUpdate(List<Request> requests,
     allDatabaseOperations.add(removePositionBatch);
     allDatabaseOperations.addAll(updateRequestsBatch);
 
-    LOG.info("Executing batch update, total records to update [{}] (including remove positions)",
+    log.info("Executing batch update, total records to update [{}] (including remove positions)",
       allDatabaseOperations.size());
 
     RequestQueueReordering payload = mapRequestsToPayload(requests);
-    LOG.info("executeRequestBatchUpdate:: instanceId: {}, requests: {}",
-      payload.getInstanceId(), payload.getRequestIds());
+    log.info("executeRequestBatchUpdate:: instanceId: {}, itemId: {}, requestLevel: {}, " +
+        "requests: {}", payload.getInstanceId(), payload.getItemId(), payload.getRequestLevel(),
+      payload.getRequestIds());
 
     batchResourceService.executeBatchUpdate(allDatabaseOperations, onFinishHandler)
       .compose(v -> eventPublisher.publishCreated(payload.getInstanceId(), payload));
   }
 
   private RequestQueueReordering mapRequestsToPayload(List<Request> requests) {
+    var firstRequest = requests.get(0);
```

Comment:
```suggestion
    var firstRequest = requests.iterator().next();
```

Reply:
Since we work with List in this method (not Collection), accessing an element by index will be more efficient for array-based implementation with constant time complexity O(1)